src/SWP/Bundle/ContentBundle/EventListener/AttachArticleToContentRouteListener.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+
+namespace SWP\Bundle\ContentBundle\EventListener;
+
+use SWP\Bundle\ContentBundle\Event\ArticleEvent;
+use SWP\Bundle\ContentBundle\Model\RouteInterface;
+use SWP\Bundle\ContentBundle\Model\RouteRepositoryInterface;
+use SWP\Bundle\CoreBundle\Model\ArticleInterface;
+
+class AttachArticleToContentRouteListener
+{
+
+    /** @var RouteRepositoryInterface  */
+    private $routeRepository;
+
+    public function __construct(RouteRepositoryInterface $routeRepository)
+    {
+
+        $this->routeRepository = $routeRepository;
+    }
+
+    public function onArticlePublish(ArticleEvent $articleEvent): void
+    {
+        /** @var ArticleInterface $article */
+        $article = $articleEvent->getArticle();
+        $route = $articleEvent->getArticle()->getRoute();
```

Comment:
You don't need to do `$articleEvent->getArticle()`, you can just use $article straight away here